grails-app/controllers/org/pih/warehouse/core/LocationController.groovy

```diff
@@ -353,22 +353,35 @@ class LocationController {
     def exportBinLocations = {
 
         Location location = Location.get(params.id)
+        Location zone = null
+
+        if (location?.isZoneLocation()) {
+            zone = location
+            location = location?.parentLocation
+        }
 
         if (!location) {
             throw new IllegalArgumentException("Must specify location")
         }
 
-        if (location.locations) {
+        def binLocations = location.internalLocations
+
+        if (zone) {
```

Comment:
Feels like this would be better off being passed to a method on Location i.e. Location.getInternalLocationsByZone(Location zone) so we only have to iterate through that internal locations once.